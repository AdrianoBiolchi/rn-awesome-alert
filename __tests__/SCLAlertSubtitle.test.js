import { SCLAlertSubtitle } from '../src/components'

import { View, Text } from 'react-native'

describe('<SCLAlertSubtitle />', () => {
  const props = {
    subtitle: 'Lorem',
    subtitleContainerStyle: {
      marginTop: 10,
    },
    subtitleStyle: {
      fontSize: 14,
    },
  }

  let wrapper
  beforeEach(() => {
    wrapper = shallow(<SCLAlertSubtitle {...props} />)
  })

  it('renders correctly', () => {
    expect(wrapper.length).toBe(1)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders the label correctly', () => {
    expect(wrapper.find(Text).contains(props.subtitle)).toBe(true)
  })

  it('renders the subtitleContainerStyle', () => {
    expect(
      wrapper
        .find(View)
        .first()
        .props().style[1],
    ).toEqual(props.subtitleContainerStyle)
  })

  it('renders the subtitleStyle', () => {
    expect(
      wrapper
        .find(Text)
        .first()
        .props().style[1],
    ).toEqual(props.subtitleStyle)
  })
})
