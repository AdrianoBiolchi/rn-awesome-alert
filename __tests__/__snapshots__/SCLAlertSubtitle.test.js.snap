// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SCLAlertSubtitle /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SCLAlertSubtitle
    subtitle="Lorem"
    subtitleContainerStyle={
      Object {
        "marginTop": 10,
      }
    }
    subtitleStyle={
      Object {
        "fontSize": 14,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={2}
        style={
          Array [
            Object {
              "color": "#555555",
              "fontSize": 34.285714285714285,
              "fontWeight": "300",
              "textAlign": "center",
            },
            Object {
              "fontSize": 14,
            },
          ]
        }
      >
        Lorem
      </Text>,
      "style": Array [
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
        Object {
          "marginTop": 10,
        },
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "accessible": true,
        "allowFontScaling": true,
        "children": "Lorem",
        "ellipsizeMode": "tail",
        "numberOfLines": 2,
        "style": Array [
          Object {
            "color": "#555555",
            "fontSize": 34.285714285714285,
            "fontWeight": "300",
            "textAlign": "center",
          },
          Object {
            "fontSize": 14,
          },
        ],
      },
      "ref": null,
      "rendered": "Lorem",
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#555555",
                "fontSize": 34.285714285714285,
                "fontWeight": "300",
                "textAlign": "center",
              },
              Object {
                "fontSize": 14,
              },
            ]
          }
        >
          Lorem
        </Text>,
        "style": Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
          },
          Object {
            "marginTop": 10,
          },
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "accessible": true,
          "allowFontScaling": true,
          "children": "Lorem",
          "ellipsizeMode": "tail",
          "numberOfLines": 2,
          "style": Array [
            Object {
              "color": "#555555",
              "fontSize": 34.285714285714285,
              "fontWeight": "300",
              "textAlign": "center",
            },
            Object {
              "fontSize": 14,
            },
          ],
        },
        "ref": null,
        "rendered": "Lorem",
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
